airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/jsonl/S3JsonlFormatConfig.java

```diff
@@ -6,35 +6,74 @@
 
 import static io.airbyte.integrations.destination.s3.S3DestinationConstants.COMPRESSION_ARG_NAME;
 import static io.airbyte.integrations.destination.s3.S3DestinationConstants.DEFAULT_COMPRESSION_TYPE;
-import static io.airbyte.integrations.destination.s3.S3DestinationConstants.FLATTEN_DATA;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.databind.JsonNode;
 import io.airbyte.integrations.destination.s3.S3Format;
 import io.airbyte.integrations.destination.s3.S3FormatConfig;
 import io.airbyte.integrations.destination.s3.util.CompressionType;
 import io.airbyte.integrations.destination.s3.util.CompressionTypeHelper;
+
 import java.util.Objects;
 
 public class S3JsonlFormatConfig implements S3FormatConfig {
 
   public static final String JSONL_SUFFIX = ".jsonl";
 
-  private final CompressionType compressionType;
+  public enum Flattening {
+
+    // These values must match the format / jsonl_flattening enum values in spec.json.
+    NO("No flattening"),
+    ROOT_LEVEL("Root level flattening");
+
+    private final String value;
+
+    Flattening(final String value) {
+      this.value = value;
+    }
 
-  private final boolean flattenData;
+    @JsonCreator
+    public static S3JsonlFormatConfig.Flattening fromValue(final String value) {
+      for (final S3JsonlFormatConfig.Flattening f : S3JsonlFormatConfig.Flattening.values()) {
+        if (f.value.equalsIgnoreCase(value)) {
+          return f;
+        }
+      }
+      throw new IllegalArgumentException("Unexpected value: " + value);
+    }
+
+    public String getValue() {
+      return value;
+    }
+
+  }
+
+  private final Flattening flattening;
+
+  private final CompressionType compressionType;
 
   public S3JsonlFormatConfig(final JsonNode formatConfig) {
-    this.compressionType = formatConfig.has(COMPRESSION_ARG_NAME)
-        ? CompressionTypeHelper.parseCompressionType(formatConfig.get(COMPRESSION_ARG_NAME))
-        : DEFAULT_COMPRESSION_TYPE;
-    this.flattenData = formatConfig.has(FLATTEN_DATA) && formatConfig.get(FLATTEN_DATA).asBoolean();
+    this(
+            Flattening.fromValue(formatConfig.has("flattening")
```

Comment:
The double ternary here makes this a little hard to read but I think that the JsonNode `get` method is safe to call without the `has` check (but would be worth testing this assumption) and the `asText` can provide a default so:
```
public S3JsonlFormatConfig(final JsonNode formatConfig) {
  this(
    formatConfig.get("flattening").asText(Flattening.NO.value),
    formatConfig.get(COMPRESSION_ARG_NAME).asText(DEFAULT_COMPRESSION_TYPE),
  );
}
```

Reply:
I was wondering about this one: wouldn't the `has` check prevent a null pointer exception in case this value does not exist in the config? I was taking the way it was done for the CSV format as an example:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/S3CsvFormatConfig.java#L55

Reply:
Yeah you're totally right, I gave you bad advice (sorry!). I was hoping `JsonNode` was smarter about this but you could still add a helper method to get the same functionality
```
private static String getOrDefault(final JsonNode formatConfig, String field, String default) {
  return formatConfig.has(field) ? formatConfig.get(field) : default;
}

public S3JsonlFormatConfig(final JsonNode formatConfig) {
  this(
    S3JsonlFormatConfig.getOrDefault(formatConfig, "flattening", Flattening.NO.value),
    S3JsonlFormatConfig.getOrDefault(formatConfig, COMPRESSION_ARG_NAME, DEFAULT_COMPRESSION_TYPE)
  );
}
```

Reply:
This is definitely a nitpick btw, non blocking

Reply:
Got it! Thank you! :D